src/services/outliningElementsCollector.ts

```diff
@@ -89,6 +93,40 @@ namespace ts.OutliningElementsCollector {
             return isFunctionBlock(node) && node.parent.kind !== SyntaxKind.ArrowFunction;
         }
 
+        function gatherRegions(): void {
+            const lineStarts = sourceFile.getLineStarts();
+
+            for (let i = 0; i < lineStarts.length; i++) {
+                const currentLineStart = lineStarts[i];
+                const lineEnd = lineStarts[i + 1] - 1 || sourceFile.getEnd();
+                const comment = sourceFile.text.substring(currentLineStart, lineEnd);
+                const result = comment.match(regionMatch);
+
+                if (result && !isInComment(sourceFile, currentLineStart)) {
+                    if (result[1] === "#region") {
+                        const start = sourceFile.getFullText().indexOf("//", currentLineStart);
+                        const textSpan = createTextSpanFromBounds(start, lineEnd);
+                        const region: OutliningSpan = {
+                            textSpan,
+                            hintSpan: textSpan,
+                            bannerText: result[2] || defaultLabel,
+                            autoCollapse: false
+                        };
+                        regions.push(region);
+                    }
+                    else {
+                        const region = regions.pop();
+                        if (region) {
+                            const newTextSpan = createTextSpanFromBounds(region.textSpan.start, lineEnd);

```

Comment:
why not just `region.textSpan.length = linEnd -region.textSpan.start;` and avoid creating the new object.